openhtf/io/output/mfg_inspector.py

```diff
@@ -334,49 +325,32 @@ def _TestRunFromTestRecord(record):
   return testrun
 
 
-class OutputToTestRunProto(object):  # pylint: disable=too-few-public-methods
+class OutputToTestRunProto(output.OutputToFile):  # pylint: disable=too-few-public-methods
   """Return an output callback that writes mfg-inspector TestRun Protos.
-
   Example filename_patterns might be:
     '/data/test_records/{dut_id}.{metadata[test_name]}.pb' or
     '/data/test_records/%(dut_id)s.%(start_time_millis)s'
-
   To use this output mechanism:
     test = openhtf.Test(PhaseOne, PhaseTwo)
     test.AddOutputCallback(openhtf.OutputToTestRunProto(
         '/data/test_records/{dut_id}.{metadata[test_name]}.pb'))
-
   Args:
     filename_pattern: A format string specifying the filename to write to,
       will be formatted with the Test Record as a dictionary.  May also be a
       file-like object to write directly to.
-
   Returns:
     filename of local file.
   """
 
   def __init__(self, filename_pattern):
-    self.filename_pattern = filename_pattern
+    super(self.__class__, self).__init__(filename_pattern)
```

Comment:
Hm, by convention we use `OutputToTestRunProto` instead of `self.__class__` here, not sure if there's any real technical justification for that.. this might act weird if someone inherits from your class, I think `self.__class__` would then be up one level and your `super()` would infinitely recurse.